lib/helper-functions.js

```diff
@@ -18,5 +21,36 @@ const componentNameToMacroName = componentName => {
 
   return `govuk${macroName}`
 }
-
 exports.componentNameToMacroName = componentNameToMacroName
+
+// Read components from various sources
+exports.SrcComponentList = fs.readdirSync(configPaths.src + 'components/')
+exports.DistComponentList = fs.readdirSync(configPaths.dist + 'components/')
+exports.PackagesComponentList = fs.readdirSync(configPaths.packages)
+
+// Compare component files from src with dist
+const compareComponentDist = (componentName) => {
```

Comment:
I wonder about the naming of this function – it's not actually doing any 'comparison', right? It's returning an object with two arrays, that you can then compare back in the test.

Should it even be one function? I think we could split it into e.g.

```
const distFilesForComponent(componentName) => {
   return fs.readdirSync(configPaths.dist + 'components/' + componentName)
}

const expectedDistFilesForComponent(componentName) => {
  let srcComponentFiles = fs.readdirSync(configPaths.src + 'components/' + componentName)
                        .filter(file => (file !== componentName + '.yaml' && file !== 'index.njk'))
  // We expect dist to contain an html sample
  srcComponentFiles.push(componentName + '.html')
  return srcComponentFiles.sort()
}
```

Then the test would be something like:
```
let actualDistFiles = lib.distFilesForComponent(componentName)
let expectedDistFiles = lib.expectedDistFilesForComponent(componentName)
expect(actualDistFiles).to.have.same.members(expectedDistFiles)
```

Which I think is clearer?

Reply:
Sounds good. Will update.